server/controllers/createSale.js

```diff
@@ -89,3 +97,33 @@ function generateSaleItems(saleRecordId, saleItems) {
   });
   return parser.insert('sale_item', saleItems);
 }
+
+function generateSubsidies(saleRecord, saleApplyableSubsidies) {
+
+  var saleApplyableSubsidiesSorted = saleApplyableSubsidies.sort(function (a, b){
+    return b.value-a.value;
+  });
+
+  var currentCost = saleRecord.cost;
+
+  var subsidyList = saleApplyableSubsidiesSorted.map(function (item) {
+    var amount;
+    if(currentCost === 0) {
+      amount = 0;
+      return  {uuid : uuid(), sale_uuid : saleRecord.uuid, subsidy_uuid : item.uuid, value : amount}
+    }else{
+      if(currentCost - item.value >= 0){
+        amount = item.value;
+      }else{
+        amount = currentCost;
+      }
+      currentCost = currentCost - amount;
+      return {uuid : uuid(), sale_uuid : saleRecord.uuid, subsidy_uuid : item.uuid, value : amount};
+    }
+  });
+
+  var saleSubsidies = subsidyList.filter(function (item){
+    return item.value > 0;
+  });
+  return parser.insert('sale_subsidy', saleSubsidies);
```

Comment:
In the past we have used parser on the server side, but doing so makes it really hard to read.  It also will be difficult to migrate to other frameworks going forward.  Finally, it takes twice as much processing on the server, since the server must 1) make a JSON object and 2) interpret the JSON object into a string with escapes and 3) execute it.

If you compose your queries as a string, the server must only concatenate and execute the string.  Is there a way you can easily transform this into a string?